### PR TITLE
QuadTree Frontend

### PR DESCRIPTION
**Pull request by anikabagga**
- Filter nearest locations by closest 1/5/10/15/20 places
- Places displayed on Google Map with markers for each location 

Note: Haven't merged by previous branch yet, so some files may be outdated. 

Screenshot 2020-07-30 at 11 02 13 AM

### CODE REVIEW

**Comment by lxcyqx** [comment]
I think the binary files should be excluded

File: capstone/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.json
```
Comment (nicorod-google) [inline capstone/.gitignore]: I'd recommend changing this to api_key.json, there might be instances where you want to include json but your excluding all of them

File: capstone/src/main/java/com/google/sps/Objects/Coordinate.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package com.google.sps.Objects;
 
-public final class Coordinate {
+import java.io.Serializable;
+
+public final class Coordinate implements Serializable{
```
Comment (nicorod-google) [inline capstone/src/main/java/com/google/sps/Objects/Coordinate.java]: nit: space after "Serializable"

File: capstone/src/main/java/com/google/sps/Objects/Location.java [hunk 3]
```diff
@@ -1,40 +1,45 @@
 package com.google.sps.Objects;
 
 import com.google.appengine.api.datastore.Entity;
+import com.google.sps.Objects.Coordinate;
+import java.io.Serializable;
 
-public final class Location {
+public final class Location implements Serializable{
```
Comment (nicorod-google) [inline capstone/src/main/java/com/google/sps/Objects/Location.java]: nit: space after "Serializable"

File: capstone/src/main/webapp/scripts/group.js [hunk 4]
```diff
@@ -1,21 +1,94 @@
...
+    locationsLimit = 1;
+  } else if (value === "5") {
+    locationsLimit = 5;
+  } else if (value === "10") {
+    locationsLimit = 10;
+  } else if(value === "15") {
+    locationsLimit = 15;
+  } else if(value === "20") {
+    locationsLimit = 20;
+  }
```
Comment (nicorod-google) [inline capstone/src/main/webapp/scripts/group.js]: You can use parseInt() to convert these to numbers instead so you don't if statements for each value you want

File: capstone/src/main/webapp/scripts/map.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  for (let i = 1; i < locationsLimit + 1; i++) {
+    addLocationMarker(allCentralGroupLocations[i]);
+  }
+  if (locationsLimit > 2) {
+    map.fitBounds(markerBounds);
+  }
+}
+
+function toggle_location_visibility() {
```
Comment (nicorod-google) [inline capstone/src/main/webapp/scripts/map.js]: nit: camel case

File: capstone/src/main/java/com/google/sps/servlets/QuadTreeServlet.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        /*NODE_CAPACITY*/ 4, 
+        /*level*/ 0, 
+        /*BoudingBox*/ new BoundingBox(25.641526, -130.429688, 50.847573, -66.269531));
+    populateQuadTree(quadTree);
+    saveState();
+  }
+
+  private void populateQuadTree(QuadTree quadTree) {
+    Scanner scanner = new Scanner(
+      getServletContext().getResourceAsStream("/WEB-INF/fast-food-location.csv"));
```
Comment (nicorod-google) [inline capstone/src/main/java/com/google/sps/servlets/QuadTreeServlet.java]: Can you add the CSV name as a constant? Will make this less dependent on this dataset and allow you to swap datasets if you wanted to

**Review by nicorod-google — CHANGES_REQUESTED** [review]

**Review by nicorod-google — APPROVED** [review]

**Review by nicorod-google — APPROVED** [review]

**Pull request merged**